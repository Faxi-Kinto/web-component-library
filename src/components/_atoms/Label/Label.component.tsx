import React from 'react';
import * as Styled from './Label.styles';

/**
 * @name Label
 *
 * @returns {JSX}
 */

export interface LabelProps {
  children?: React.ReactNode;
  className?: string;
  htmlFor?: string;
}

const Label: React.FC<LabelProps> = (props: LabelProps): JSX.Element => {
  const { children, className, htmlFor } = props;
  return (
    <Styled.Label htmlFor={htmlFor} className={className}>
      {children}
    </Styled.Label>
  );
};

export default Label;
