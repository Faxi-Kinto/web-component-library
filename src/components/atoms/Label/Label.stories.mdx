import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import { action } from '@storybook/addon-actions';
import Button from "./";
import Icon from "../Icon"

<Meta title="Core|Atoms/Button" />

# Button Component
This is a fundemental atom used to indicate a call to action for the user. It can take two forms that emphasize its hierarchy. 

## Implementation

#### Default
<Preview>
  <Story name="default">
    <Button onClick={action('button-click')}>
      <p>Hello from Button</p>
    </Button>
  </Story>
</Preview>

#### Secondary
<Preview>
  <Story name="secondary">
    <Button secondary>
      <p>Exit without saving</p><Icon name="times" color="#00708D"/>
    </Button>
  </Story>
</Preview>

# Props

<Props of={Button} />
