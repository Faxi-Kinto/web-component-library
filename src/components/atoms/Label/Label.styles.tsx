import styled from 'styled-components';
import { pxToRem } from '@faxi/web-css-utilities';

export const Label = styled.label`
  display: inline-flex;
  height: ${pxToRem('24px')};
  align-items: flex-end;
  margin-bottom: ${pxToRem('5px')};
`;
