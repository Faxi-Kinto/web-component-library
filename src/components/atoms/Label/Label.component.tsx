import React from 'react';
import * as Styled from './Label.styles';

/**
 * @name Label
 *
 * [Insert Component Description]
 *
 * @returns {JSX}
 */

export interface LabelProps {
  children?: React.ReactNode;
  htmlFor?: string;
  className?: string;
}

const Label: React.FC<LabelProps> = (props: LabelProps): JSX.Element => {
  const { children, htmlFor, className } = props;
  return (
    <Styled.Label htmlFor={htmlFor} className={className}>
      {children}
    </Styled.Label>
  );
};

export default Label;
