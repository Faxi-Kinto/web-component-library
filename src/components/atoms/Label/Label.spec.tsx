import React from 'react';
import renderer from 'react-test-renderer';
import Label from './Label.component';

test('Label renders correctly', () => {
  const component = renderer.create(<Label>Label</Label>);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
