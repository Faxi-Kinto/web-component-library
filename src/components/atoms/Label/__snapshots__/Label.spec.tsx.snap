// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
.c0 {
  color: #4E606A;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1.5rem;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-bottom: 0.3125rem;
}

<label
  className="c0"
>
  Label
</label>
`;
